### DISCUSSION TITLE
Why is `Depends` function not typesafe? (easy fix?)

### DISCUSSION THREAD

**Discussion by gregpr07** [fastapi/fastapi#9872] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# instead of default `Depends function`

def Depends(  # noqa: N802
    dependency: Optional[Callable[..., Any]] = None, *, use_cache: bool = True
) -> Any:
    return params.Depends(dependency=dependency, use_cache=use_cache)
```

# it could be just 
```
def SafeDepends(dependency: Callable[..., Coroutine[Any, Any, T]], *, use_cache: bool = True) -> T:  # noqa: N802
    return cast(T, params.Depends(dependency=dependency, use_cache=use_cache))
```
```


### Description

Make Depends typesafe

### Operating System

macOS

### Operating System Details

_No response_

### FastAPI Version

0.98.0

### Python Version

3.10.0

### Additional Context

_No response_

**Comment by Kludex** [comment, marked as answer]
"it could be just" - it cannot. There are cases it doesn't cover.

You can use Annotated on these cases. See https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/#type-annotation-vs-depends for more details.
